Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewHelpers.fs

```diff
@@ -14,11 +14,13 @@ module ViewHelpers =
     let identical (x: 'T) (y:'T) = System.Object.ReferenceEquals(x, y)
     
     let getKey (el:ViewElement) = el.TryGetKey()
-          
+    
+    let sameType (prevChild: ViewElement) (newChild: ViewElement) =
+        prevChild.TargetType = newChild.TargetType
```

Comment:
Don't add this function, use `canReuse` instead.
`canReuse` already does this check along with other mandatory checks to make sure we can really update a control.

Reply:
Ok, so testcases need to be updated at least same automationId should be set

Reply:
Which tests?
There should be no need for automationId in my opinion.

Reply:
canReuse relies on automationId in its checks

Reply:
Yes, it only checks that the AutomationId didn't changed since XF prohibits changing an already set AutomationId (you can go from nothing to a value, but not from a value to another value).

So since the tests never define an AutomationId, it won't influence the result of canReuse (if it's returning false, it's another reason).

Reply:
Yes, you are right